dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/test/package-info.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * This package contains DHIS2 related test code. Only add code that is reusable across modules.
+ *
+ * <p>This guide helps you set up your test.
+ *
+ * <ul>
+ *   <li>Do you need a DB to test your code?
+ *       <ul>
+ *         <li>No: write a unit test.
+ *         <li>Yes: move on to the next question.
+ *       </ul>
+ *   <li>Do you need to test a controller?
+ *       <ul>
+ *         <li>Yes: create a test class in the maven module {@code dhis-test-web-api} extending one
+ *             of {@code org.hisp.dhis.test.webapi.PostgresControllerIntegrationTestBase} or {@code
+ *             org.hisp.dhis.test.webapi.H2ControllerIntegrationTestBase} depending on the DB you
+ *             want to test against.
+ *         <li>No: create a test class in the maven module {@code dhis-test-integration} extending
+ *             {@code org.hisp.dhis.test.integration.PostgresIntegrationTestBase}.
+ *       </ul>
+ *   <li>Do you need to test code running in another thread?
+ *       <ul>
+ *         <li>Yes: <b>do not</b> annotate your test class with {@link
+ *             org.springframework.transaction.annotation.Transactional} so that the changes to the
+ *             DB during testing are visible in other threads. {@link
+ *             org.hisp.dhis.test.junit.SpringIntegrationTestExtension} will do the DB cleanup.
+ *         <li>No: annotate your test class with {@link
+ *             org.springframework.transaction.annotation.Transactional} so that the changes to the
+ *             DB during testing are rolled back automatically by Spring.
+ *       </ul>
+ *   <li>Can you setup your test data once? By default JUnit tests have a {@link
```

Comment:
I am confused by the phrase here.
@BeforeEach annotation will work the same no matter what is the Lifecycle used, right?